packages/adblocker/src/lists.ts

```diff
@@ -251,40 +318,152 @@ export function generateDiff(
   // Set config to `debug` so that we keep track of raw lines for each filter
   const debugConfig = new Config(Object.assign({}, config, { debug: true }));
 
-  const prevRevisionFilters = getFilters(prevRevision, debugConfig);
-  const prevRevisionIds = new Set(prevRevisionFilters.map((filter) => filter.getId()));
+  const prevRevisionData = getFilters(prevRevision, debugConfig);
+  const prevRevisionIds = new Set(prevRevisionData.filters.map((filter) => filter.getId()));
+  const prevRevisionPreprocessorAffectedFilterIds = prevRevisionData.preprocessors.reduce(
+    (state, preprocessor) => {
+      for (const filterID of preprocessor.filterIDs) {
+        state.add(filterID);
+      }
+
+      return state;
+    },
+    new Set<number>(),
+  );
+
+  const newRevisionData = getFilters(newRevision, debugConfig);
+  const newRevisionIds = new Set(newRevisionData.filters.map((filter) => filter.getId()));
+  const newRevisionPreprocessorAffectedFilterIds = newRevisionData.preprocessors.reduce(
+    (state, preprocessor) => {
+      for (const filterID of preprocessor.filterIDs) {
+        state.add(filterID);
+      }
+
+      return state;
+    },
+    new Set<number>(),
+  );
 
-  const newRevisionFilters = getFilters(newRevision, debugConfig);
-  const newRevisionIds = new Set(newRevisionFilters.map((filter) => filter.getId()));
+  const index: Map<number, string> = new Map();
 
   // Check which filters were added, based on ID
   const added: Set<string> = new Set();
-  for (const filter of newRevisionFilters) {
-    if (!prevRevisionIds.has(filter.getId())) {
+  for (const filter of newRevisionData.filters) {
+    if (
+      !prevRevisionIds.has(filter.getId()) &&
+      !newRevisionPreprocessorAffectedFilterIds.has(filter.getId())
+    ) {
       added.add(filter.rawLine as string);
     }
+
+    index.set(filter.getId(), filter.rawLine as string);
   }
 
   // Check which filters were removed, based on ID
   const removed: Set<string> = new Set();
-  for (const filter of prevRevisionFilters) {
-    if (!newRevisionIds.has(filter.getId())) {
+  for (const filter of prevRevisionData.filters) {
+    if (
+      !newRevisionIds.has(filter.getId()) &&
+      !prevRevisionPreprocessorAffectedFilterIds.has(filter.getId())
+    ) {
       removed.add(filter.rawLine as string);
     }
+
+    index.set(filter.getId(), filter.rawLine as string);
+  }
+
+  // Fast exit if we don't want to handle preprocessors.
+  if (!config.loadPreprocessors) {
```

Comment:
IMO if preprocessors are disabled we should not have any preprocessor-related logic above that line (i.e. `prevRevisionPreprocessorAffectedFilterIds` and `newRevisionPreprocessorAffectedFilterIds` should only be created if `loadPreprocessors` is `true`)